gtsfm/utils/metrics.py

```diff
@@ -167,19 +167,71 @@ def compute_averaging_metrics(
         else:
             wTi_list.append(Pose3(wRi, wti))
 
-    # ground truth is the reference/target for alignment
-    wTi_aligned_list = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+    # ground truth is the reference/target for alignment. discard 2nd return arg -- the estimated Similarity(3) object
+    wTi_aligned_list, _ = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+
+    i2Ui1_dict_gt = simulate_twoview_translation_direction_measurements(gt_wTi_list)
 
     wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(wTi_aligned_list)
     gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(gt_wTi_list)
 
     metrics = []
     metrics.append(compute_rotation_angle_metric(wRi_aligned_list, gt_wRi_list))
     metrics.append(compute_translation_distance_metric(wti_aligned_list, gt_wti_list))
-    metrics.append(compute_translation_angle_metric(i2Ui1_dict, wTi_aligned_list))
+    metrics.append(compute_translation_angle_metric(i2Ui1_dict=i2Ui1_dict_gt, wTi_list=wTi_aligned_list))
     return GtsfmMetricsGroup(name="averaging_metrics", metrics=metrics)
 
 
+def compute_ba_pose_metrics(
+    gt_wTi_list: List[Pose3],
+    ba_output: GtsfmData,
+) -> GtsfmMetricsGroup:
+    """Compute pose errors w.r.t. GT for the bundle adjustment result.
+
+    Note: inputs must be aligned beforehand to the ground truth.
+
+    Args:
+        gt_wTi_list: List of ground truth poses.
+        ba_output: sparse multi-view result, as output of bundle adjustment.
+
+    Returns:
+        A group of metrics that describe errors associated with a bundle adjustment result (w.r.t. GT).
+    """
+    wTi_aligned_list = ba_output.get_camera_poses()
+    i2Ui1_dict_gt = simulate_twoview_translation_direction_measurements(gt_wTi_list)
+    
+    wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(wTi_aligned_list)
+    gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(gt_wTi_list)
+
+    metrics = []
+    metrics.append(compute_rotation_angle_metric(wRi_aligned_list, gt_wRi_list))
+    metrics.append(compute_translation_distance_metric(wti_aligned_list, gt_wti_list))
+    metrics.append(compute_translation_angle_metric(i2Ui1_dict_gt, wTi_aligned_list))
+    return GtsfmMetricsGroup(name="mvs_input_metrics", metrics=metrics)
+
+
+def simulate_twoview_translation_direction_measurements(gt_wTi_list: List[Pose3]) -> Dict[Tuple[int, int], Unit3]:
```

Comment:
good point, let me update the name.

I'm not sure which module would be better for it -- any ideas?

Reply:
Not sure which module is better, we can leave it here for now. Also, could you update the argument name? The function has nothing to do with `gt_` so that can be removed.

Reply:
updated